### ISSUE TITLE
SpeedMonitoring:SpeedSupervision throws EB

### ISSUE THREAD

**Issue by BerndHekele** [openETCS/modeling#839]
We get an EB request from SDM.

![image](https://cloud.githubusercontent.com/assets/3288918/10607405/9acfcd08-7739-11e5-89ad-c664e09b4986.png)

Train is running in NTC.
The use of the EB is depending on Mode / Level in this track-scenario.
I propse to shortcut this sort of check in level ntc.

I will contin ue testing with a local shortcut.

**Comment by T12z** [comment]
i will fix later, but obviously trainPositionUnknown == true is rubbish.
Sorry for that!

On 20.10.2015 14:53, Bernd Hekele wrote:
>
> We get an EB request from SDM.
>
> image 
> <https://cloud.githubusercontent.com/assets/3288918/10607405/9acfcd08-7739-11e5-89ad-c664e09b4986.png>
>
> Train is running in NTC.
> The use of the EB is depending on Mode / Level in this track-scenario.
> I propse to shortcut this sort of check in level ntc.
>
> I will contin ue testing with a local shortcut.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/openETCS/modeling/issues/839>.
>

-- 
Dipl.-Ing. Thorsten Schulz
Universität Rostock, Fakultät f. Informatik und Elektrotechnik
Institut f. Angewandte Mikroelektronik und Datentechnik
18051 Rostock, Germany
+49 381 498-7278
http://www.imd.uni-rostock.de

**Comment by T12z** [comment]
Fixed in last commit.
Check has been disabled, until I am sure when I can do this check. Anyways, shortcut in NTC is not possible, as SDM does not know it's NTC. The TA fakes all information as if in FS.

Frankly no idea, why it did not kick when I tested the model yesterday. Maybe because odometry was foobared because of the miscalculation in the simulation :construction_worker: 

On the long run, I would rather raise the exception to an output instead of EB and let some other logic decide how to act - but that's another interface change and some kind exception-catch architecture needs to be implemented. just a rant.
